Integrations/integration-AzureCompute.yml

```diff
@@ -0,0 +1,1085 @@
+
+        returns:
+            JSON Response Object
+        """
+        try:
+            url = full_url if full_url else BASE_URL + url_suffix
+            r = requests.request(
+                method,
+                url,
+                headers=headers,
+                data=data,
+                verify=USE_SSL,
+                json=j_son
+            )
+            green_codes = codes if codes else {200, 201, 202, 204}
+            if r.status_code not in green_codes:
+                err_msg = 'Error in API call to Azure Compute Integration [{}] - {}'.format(r.status_code, r.reason)
+                err = r.json().get('error')
+                if err:
+                    err_msg1 = "\nError code: {}\nError message: {}".format(err.get('code'), err.get('message'))
+                    err_msg += err_msg1
+                raise Exception(err_msg)
+            response = json.loads(r.content)
+        except ValueError as e:
+            response = r.content
+        except Exception as e:
+            raise
+
+        return response
+
+
+    '''MAIN FUNCTIONS / API CALLS'''
+
+    #<-------- Resource Groups -------->#
+
+    def list_resource_groups():
+        endpoint_url = 'https://management.azure.com/subscriptions/' + SUBSCRIPTION_ID
+        endpoint_url += '/resourcegroups?api-version=2018-05-01'
+        update_access_token()
+        response = http_request('GET', full_url=endpoint_url, codes={200})
+        return response
+
+
+    def list_resource_groups_command():
+        """
+        List all Resource Groups belonging to your Azure subscription
+
+        returns:
+            Resource-Group Objects
+        """
+        response = list_resource_groups()
+        # Retrieve relevant properties to return to context
+        value = response.get('value')
+        resource_groups = []
+        for resource_group in value:
+            rg = {
+                'Name': resource_group.get('name'),
+                'ID': resource_group.get('id'),
+                'Location': resource_group.get('location'),
+                'ProvisioningState': resource_group.get('properties').get('provisioningState')
+            }
+            resource_groups.append(rg)
+
+        title = "List of Resource Groups"
+        human_readable = tableToMarkdown(title, resource_groups)
+        demisto.results({
+            'Type': entryTypes['note'],
+            'Contents': response,
+            'ContentsFormat': formats['text'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': human_readable,
+            'EntryContext': {'Azure.ResourceGroup(val.Name && val.Name === obj.Name)': resource_groups}
+        })
+
+
+    #<-------- Virtual Machines -------->#
+
+    def list_vms(resource_group):
+        # Construct endpoint URI suffix
+        url_endpoint = resource_group + '/providers/Microsoft.Compute/virtualMachines?api-version=' + API_VERSION
```

Comment:
can't we append the api version in the http request?

Reply:
I made it part of the passed parameters as you noted elsewhere in your code review